src/main/java/duke/task/Ui.java

```diff
@@ -0,0 +1,63 @@
+package duke.task;
+
+/**
+ * Encapsulates an Ui object that interacts with the user.
+ */
+public class Ui {
+
+    protected static TaskList tasks;
+
+    Ui(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    protected static String showWelcome() {
+        return "Hello! I'm Yoo ( ^-^)/ \nWhat can I do for you?";
+    }
+
+    protected static String showWelcomeTaskList() {
```

Comment:
showWelcomeTaskList can make use of showTaskList() for ease of change in future, or it might be better to collapse it into a single method since they are somewhat similar.